docs/content/guides/dagster/pythonic-resources.mdx

```diff
@@ -379,8 +375,69 @@ def my_asset(writer: Writer):
     writer.output("hello, world!")
 
 
+defs = Definitions(assets=[my_asset], resources={"writer": WriterResource(prefix="greeting: ")})
+```
+
+### Adapting function-style I/O managers
+
+Similar to with resources, in codebases that utilize [function-style](https://docs.dagster.io/concepts/io-management/io-managers) I/O managers with the <PyObject object="@io_manager"/> decorator, you can use the <PyObject object="ConfigurableIOManagerAdapter"/> class to adapt them to the new Pythonic I/O manager pattern.
+
+The adapter subclass should list all config fields as attributes and implement a `wrapped_io_manager` property which returns the underlying I/O manager definition to adapt.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_io_adapter endbefore=end_io_adapter
+from dagster import (
+    Definitions,
+    IOManagerDefinition,
+    io_manager,
+    IOManager,
+    InputContext,
+    OutputContext,
+)
+from dagster._config.structured_config import (
+    StructuredIOManagerAdapter as ConfigurableIOManagerAdapter,
+)
+import os
+
+
+# Old code, cannot be changed for back-compat purposes
```

Comment:
I think one modification which should make here is that that we can't change the public interface for the code, rather than not change it at all.